lib/gui/models/drives.js

```diff
@@ -20,103 +20,94 @@
+ * @throws Will throw if drives is not an array of objects
+ *
+ * @example
+ * availableDrives.setDrives([ ... ]);
+ */
+exports.setDrives = (drives) => {
+  Store.dispatch({
+    type: Store.Actions.SET_AVAILABLE_DRIVES,
+    data: drives
+  });
+};
 
-  /**
-   * @summary Memoize ImmutableJS list reference
-   * @function
-   * @private
-   *
-   * @description
-   * This workaround is needed to avoid AngularJS from getting
-   * caught in an infinite digest loop when using `ngRepeat`
-   * over a function that returns a mutable version of an
-   * ImmutableJS object.
-   *
-   * The problem is that every time you call `myImmutableObject.toJS()`
-   * you will get a new object, whose reference is different from
-   * the one you previously got, even if the data is exactly the same.
-   *
-   * @param {Function} func - function that returns an ImmutableJS list
-   * @returns {Function} memoized function
-   *
-   * @example
-   * const getList = () => {
-   *   return Store.getState().toJS().myList;
-   * };
-   *
-   * const memoizedFunction = memoizeImmutableListReference(getList);
-   */
-  const memoizeImmutableListReference = (func) => {
-    let previous = [];
+/**
+ * @summary Memoize ImmutableJS list reference
+ * @function
+ * @private
+ *
+ * @description
+ * This workaround is needed to avoid AngularJS from getting
+ * caught in an infinite digest loop when using `ngRepeat`
+ * over a function that returns a mutable version of an
+ * ImmutableJS object.
+ *
+ * The problem is that every time you call `myImmutableObject.toJS()`
+ * you will get a new object, whose reference is different from
+ * the one you previously got, even if the data is exactly the same.
+ *
+ * @param {Function} func - function that returns an ImmutableJS list
+ * @returns {Function} memoized function
+ *
+ * @example
+ * const getList = () => {
+ *   return Store.getState().toJS().myList;
+ * };
+ *
+ * const memoizedFunction = memoizeImmutableListReference(getList);
+ */
+const memoizeImmutableListReference = (func) => {
+  let previous = [];
 
-    return (...args) => {
-      const list = Reflect.apply(func, this, args);
+  return (...args) => {
+    const list = Reflect.apply(func, this, args);
 
-      if (!_.isEqual(list, previous)) {
-        previous = list;
-      }
+    if (!_.isEqual(list, previous)) {
+      previous = list;
+    }
 
-      return previous;
-    };
+    return previous;
   };
+};
 
-  /**
-   * @summary Get detected drives
-   * @function
-   * @private
-   *
-   * @returns {Object[]} drives
-   *
-   * @example
-   * const drives = DrivesModel.getDrives();
-   */
-  this.getDrives = memoizeImmutableListReference(() => {
-    return Store.getState().toJS().availableDrives;
-  });
-
+/**
+ * @summary Get detected drives
+ * @function
+ * @private
+ *
+ * @returns {Object[]} drives
+ *
+ * @example
+ * const drives = availableDrives.getDrives();
+ */
+exports.getDrives = memoizeImmutableListReference(() => {
```

Comment:
I believe this whole `memoizeImmutableListReference` hack only applies to the usage of this function in the `drive-selector` component. If so, could we wrap `.getDrives()` and apply `memoizeImmutableListReference` over there instead?